Domain/sdLitica.Events/Bus/EventRegistry.cs

```diff
@@ -3,44 +3,45 @@
 using System.Collections.Generic;
 using System.Text;
 using System.Linq;
+using sdLitica.Messages.Abstractions;
 
 namespace sdLitica.Events.Bus
 {
     public class EventRegistry : IEventRegistry
     {
         private readonly IDictionary<Type, IList<string>> _eventRegistry;
-                   
-        public EventRegistry()
+        private readonly IBrokerManager _brokerManager;
+
+        public EventRegistry(IBrokerManager brokerManager)
         {
             _eventRegistry = new Dictionary<Type, IList<string>>();
+            _brokerManager = brokerManager;
         }
 
-        public void Register<T>(string exchange) where T : IEvent
+        public void Register<T>(string queue) where T : IEvent
         {
             var eventType = typeof(T);
 
             if (_eventRegistry.ContainsKey(eventType))
             {
-                if (!_eventRegistry[eventType].Contains(exchange))
-                    _eventRegistry[eventType].Add(exchange);
+                if (!_eventRegistry[eventType].Contains(queue))
+                    _eventRegistry[eventType].Add(queue);
                 return;
             }
 
-            _eventRegistry.Add(new KeyValuePair<Type, IList<string>>(eventType, new List<string>() { exchange }));
+            _eventRegistry.Add(new KeyValuePair<Type, IList<string>>(eventType, new List<string>() { queue }));
+            
+            //TODO needs: create queue OR exchanges
+            _brokerManager.CreateQueue(queue);
         }
 
```

Comment:
"public void Register<T>" registers queues, while "public IList<string> GetExchangesForEvent<T>" registers exchanges. 
Also, wouldn't be better to use Topic exchanges, rather than publish one event to many Direct exchanges?

Reply:
You are right. `GetExchangesForEvent`  is not clear. It remain from previous implementation. This should be `GetQueuesForEvents`, or just `GetForEvent`. The code supports publishing one event to many queues, yes. The code must support in future to publish to a Topic. I did not have time to code both cases. The defined pub/sub architecture can select to use one or another approach, depending on each case (queue or topic). It is important that the code support both. Feel free to code the publishing to a Topic also. Thanks.